nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/GetAzureEventHub.java

```diff
@@ -70,7 +74,6 @@
         @WritesAttribute(attribute = "eventhub.partition", description = "The name of the Azure Partition from which the message was pulled")
 })
 public class GetAzureEventHub extends AbstractProcessor {
```

Comment:
this processor I suspect will be rarely used in comparison to ConsumeAzureEVentHub anyway since this doesn't utilize the record reader/writer mechanism.  The other while using a single thread as currently implemented will probably offer a better user experience and performance characteristic.  If necessary that one could be updated to have multiple threads operating at once.  This one will have multiple threads but writing of data not being record oriented will mean overall throughput will drag.  This is the same reason we went from ConsumeKafka to ConsumeKafkaRecord design.  So with that in mind I think we can stress a bit less on this one.  As long is it functions correctly, the thread pool is understood by the user, and is destroyed correctly - then this is prob fine

Reply:
Yes we also found that the GetAzureEventHub processor serves similar functionality as ConsumeAzureEventHub. This GetAzureEventHub processor uses **EventHubClient**, and can allow user to receive from specific partition id. However, currently it does not let user specify which partition to read from. Instead, it just traverse through all partitions in a round robin way, so it works same functionality as ConsumeAzureEventHub, which uses **EPH**. This can be a next step to improve, and provide customers with more convinient and flexible control.